src/pe/data_directories.rs

```diff
@@ -103,4 +213,14 @@ impl DataDirectories {
         let idx = 14;
         &self.data_directories[idx]
     }
+
+    pub fn dirs(&self) -> impl Iterator<Item = (DataDirectoryType, DataDirectory)> {
+        self.data_directories
+            .into_iter()
+            .enumerate()
+            // (Index, Option<DD>) -> Option<(Index, DD)> -> (DDT, DD)
+            .filter_map(|(i, o)|
+                // We should not have invalid indexes.
+                o.map(|v| (i.try_into().unwrap(), v)))
```

Comment:
i don't like the unwrap, are you sure it won't fail?

Reply:
The data directories is an fixed `[Option<DataDirectory>; 15]`, when you do enumerate on it
you will get `(0, opt), …, (15, opt)` necessarily, given that, the `TryFrom` function above will match over the value and send it to a valid value and only fail if the index number is 16 or more.

Because of this, it's impossible to have a failure for this `unwrap` as long as Rust doesn't guarantee the structure and the type of `data_directories`.

I can add this comment to prove the safety of the unwrap here.